### PR TITLE
rfc: add new pipeline

### PR DESCRIPTION
**Pull request by yorkie**
See the markdown to review.

### CODE REVIEW

**Comment by yorkie** [comment]
I will add JSON schema later, please have reviews on other parts.

**Comment by yorkie** [comment]
Added JSON schema definition, @WenheLI @rickycao-qy.

**Comment by yorkie** [comment]
@rickycao-qy May I ask you to update the RL parts on this proposal?

File: docs/rfcs/0000-new-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      "params": {
+        "epochs": 200
+      }
+    },
+    {
+      "type": "modelEvaluate",
+      "package": "@pipcook/plugins-tfjs-text-lstm-model-evaluate"
+    },
+    {
+      "type": "deploy",
```
Comment (rickycao-qy) [inline docs/rfcs/0000-new-pipeline.md:61]: Is it better to specify the type of 'modelDeploy', just to be in conformity with other plugins' types

File: docs/rfcs/0000-new-pipeline.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+- multiple nodes for the same plugin.
+- model deployment and distribution.
+
+# Basic example
+
+```json
+{
+  "specVersion": "2.0",
+  "plugins": [
```
Comment (rickycao-qy) [inline docs/rfcs/0000-new-pipeline.md:19]: what if the user specifies more than one dataCollect/dataAccess/modelDeinfe plugins? Shall we make the whole plugins to be an array or just dataProcess to be an array?

File: docs/rfcs/0000-new-pipeline.md (hunk 1 above)
Comment (yorkie) [reply docs/rfcs/0000-new-pipeline.md:61]: The name "modelDeploy" is the deprecated type, I use the new and common name is that what the plugin to deploy is not only the model, includes the runtime, data processing, and others.

File: docs/rfcs/0000-new-pipeline.md (hunk 2 above)
Comment (yorkie) [reply docs/rfcs/0000-new-pipeline.md:19]: To make sure the extensibility of plugins, I think we could keep all plugins should be in the array, how about add additional rules to restrict the array such as limit only 1 "modelDefine" type?

Comment (rickycao-qy) [reply docs/rfcs/0000-new-pipeline.md:19]: This sounds good. We can modify the schema so that the numbers are limited to 1 for now.

Comment (yorkie) [reply docs/rfcs/0000-new-pipeline.md:19]: @rickycao-qy I have updated the schema below, the plugin items are split into multiple sub-schemas, which I defined the `maxItems` and `minItems` in different ones.

**Comment by yorkie** [comment]
Within the separation of plugin items, I also defined some of the common parameters for every specific type of plugin, for example,

- added `url` to the params once the plugin type is "dataCollect"
- added `epochs` to the params once the plugin type is "modelTrain"
- ...

And we need to make sure these names are consistent with our core interfaces for plugin developers.

**Comment by rickycao-qy** [comment]
> Within the separation of plugin items, I also defined some of the common parameters for every specific type of plugin, for example,
> 
> * added `url` to the params once the plugin type is "dataCollect"
> * added `epochs` to the params once the plugin type is "modelTrain"
> * ...
> 
> And we need to make sure these names are consistent with our core interfaces for plugin developers.

Just wondering if it's necessary to add these common parameters cuz the user may implement specific data-collect plugin just to collect specific data without url.

**Comment by yorkie** [comment]
How about making the URL in `data-collect` be optional?

**Comment by rickycao-qy** [comment]
> How about making the URL in `data-collect` be optional?

Sounds good to me

**Comment by yorkie** [comment]
> > How about making the URL in `data-collect` be optional?
> 
> Sounds good to me

Updated :)

**Review by rickycao-qy — APPROVED** [review]
LGTM

**Review by FeelyChau — APPROVED** [review]
LGTM.

**Review by WenheLI — APPROVED** [review]
LGTM

**Pull request merged**